src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKeyProvider.java

```diff
@@ -270,6 +537,20 @@ private static Region getStartingRegion(final String keyArn) {
         return Region.getRegion(Regions.DEFAULT_REGION);
     }
 
+    private String identifyKeyRegion(final String keyArn) {
+        String region = parseRegionfromKeyArn(keyArn);
+
+        if (region != null) {
+            return region;
+        }
+
+        if (defaultRegion_ == null) {
+            throw new AwsCryptoException("Can't use non-ARN key identifiers or aliases when no default region is set");
```

Comment:
We should probably complain on `build()` actually - since the list of key IDs for encrypt can't change after construction using the builder, and this isn't relevant on decrypt.

Reply:
++